com.unity.sg2/Editor/GraphDeltaRegistry/Registry/NodeUIInfo.cs

```diff
@@ -3,22 +3,36 @@
 
 namespace UnityEditor.ShaderGraph.GraphDelta
 {
+    interface INodeUIDescriptorBuilder
+    {
+        NodeUIDescriptor CreateDescriptor(NodeHandler handler);
+    }
+
+    internal class StaticNodeUIDescriptorBuilder : INodeUIDescriptorBuilder
+    {
+        private NodeUIDescriptor descriptor;
+        public StaticNodeUIDescriptorBuilder(NodeUIDescriptor descriptor) => this.descriptor = descriptor;
+        public NodeUIDescriptor CreateDescriptor(NodeHandler handler) => descriptor;
```

Comment:
Is this function taking in a NodeHandler instance but not using it intentional? I recall that idea you mentioned of associating a node with multiple registered ui descriptors so maybe something to do with that?

Reply:
It's part of an interface that would allow the NodeUIDescriptor to be built reactively against the data in the NodeHandler. This implementation of the interface is 'static,' and used in the ShaderGraphRegistry for wrapping when a NodeUIDescriptor is registered directly.